agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AiConfigCustomizer.java

```diff
@@ -53,22 +53,41 @@ public Map<String, String> apply(ConfigProperties otelConfig) {
         configuration.preview.captureHttpClientHeaders.responseHeaders);
 
     // enable capturing all mdc properties
-    properties.put(
-        "otel.instrumentation.logback-appender.experimental.capture-mdc-attributes", "*");
-    properties.put("otel.instrumentation.log4j-appender.experimental.capture-mdc-attributes", "*");
-    properties.put(
-        "otel.instrumentation.log4j-appender.experimental.capture-context-data-attributes", "*");
-    properties.put(
-        "otel.instrumentation.jboss-logmanager.experimental.capture-mdc-attributes", "*");
+    if (configuration.instrumentation.log4jAppender.enabled) {
+      properties.put(
+          "otel.instrumentation.log4j-appender.experimental.capture-mdc-attributes", "*");
+      properties.put(
+          "otel.instrumentation.log4j-appender.experimental.capture-context-data-attributes", "*");
+      properties.put(
+          "otel.instrumentation.log4j-appender.experimental.capture-map-message-attributes",
+          "true");
+    } else {
+      properties.put(
+          "otel.instrumentation.log4j-appender.experimental.capture-map-message-attributes",
+          "false");
+    }
 
+    if (configuration.instrumentation.logbackAppender.enabled) {
```

Comment:
Coould you gather together all the `if(configuration.instrumentation.logbackAppender.enabled)`?

Reply:
same here